alf/algorithms/particle_vi_algorithm_test.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) 2020 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+
+from absl import logging
+from absl.testing import parameterized
+import torch
+import torch.nn as nn
+
+import alf
+from alf.algorithms.particle_vi_algorithm import ParVIAlgorithm
+from alf.networks import Network
+from alf.tensor_specs import TensorSpec
+
+
+class ParVIAlgorithmTest(parameterized.TestCase, alf.test.TestCase):
+    def assertArrayEqual(self, x, y, eps):
+        self.assertEqual(x.shape, y.shape)
+        self.assertLessEqual(float(torch.max(abs(x - y))), eps)
+
+    def cov(self, data, rowvar=False):
+        """Estimate a covariance matrix given data.
+
+        Args:
+            data (tensor): A 1-D or 2-D tensor containing multiple observations 
+                of multiple dimentions. Each row of ``mat`` represents a
+                dimension of the observation, and each column a single
+                observation.
+            rowvar (bool): If True, then each row represents a dimension, with
+                observations in the columns. Othewise, each column represents
+                a dimension while the rows contains observations.
+
+        Returns:
+            The covariance matrix
+        """
+        x = data.detach().clone()
+        if x.dim() > 2:
+            raise ValueError('data has more than 2 dimensions')
+        if x.dim() < 2:
+            x = x.view(1, -1)
+        if not rowvar and x.size(0) != 1:
+            x = x.t()
+        fact = 1.0 / (x.size(1) - 1)
+        x -= torch.mean(x, dim=1, keepdim=True)
+        return fact * x.matmul(x.t()).squeeze()
+
+    @parameterized.parameters(('svgd'), ('gfsf'))
+    def test_par_vi_algorithm(self, par_vi='svgd'):
+        """
+        The generator is trained to match(STEIN)/maximize(ML) the likelihood
```

Comment:
Need to update the comment. It's not generator. And there no "maximize"

Reply:
Updated.